source/nsdynmemLIB/nsdynmemLIB.c

```diff
@@ -133,14 +133,15 @@ static int heap_alloc_internal_check(int16_t alloc_size)
 
     if (alloc_size) {
         //
-        temp_sector = alloc_size;
-        temp_int = (temp_sector % temp_size);
-        if (temp_int) {
-            temp_sector += (temp_size - temp_int);
-        }
+        temp_sector = (alloc_size+temp_size-1)/temp_size;
```

Comment:
Maths looks correct.
Lose the "temp_size" - just use "sizeof(int)" - it will be clearer.
Remove the commented code.
Get the formatting right - spaces around operators.